core/build.gradle

```diff
@@ -24,7 +24,10 @@ buildProperties {
     publish {
         def generateVersion = {
             boolean isSnapshot = cli['bintraySnapshot'].or(false).boolean
-            return isSnapshot ? "SNAPSHOT-${System.getenv('BUILD_NUMBER')?:'LOCAL'}" : version
+            boolean isExperimental = cli['bintrayExperimental'].or(false).boolean
+            def snapshotVersion = "SNAPSHOT-${System.getenv('BUILD_NUMBER') ?: 'LOCAL'}"
+            def experimentalVersion = "EXPERIMENTAL-${System.getenv('BUILD_NUMBER') ?: 'LOCAL'}"
+            return isSnapshot ? snapshotVersion : isExperimental ? experimentalVersion : version
```

Comment:
I don't dislike it, but I don't love it either. Pretty sure this can be re-written in a more elegant (aka less confusing) way, just not sure how 😅 
Maybe @devisnik has some idea?

Reply:
Hm, I think it's either what you have or 
````groovy
if (...) 
else if (...)
else 
```

This more verbose control structure might allow you to inline all the local variables, so it might not be that bad in terms of lines of code.
I'm not aware of something in the middle, which I guess you are looking for here. Sorry.

Reply:
What about a when with no arguments ?

```
when {
   isSnapshot -> snapshotVersion 
   isExperimental -> experimentalVersion
   default -> version
} 
```

Reply:
```
if(isSnapshot) {
 return "SNAPSHOT-${System.getenv('BUILD_NUMBER') ?: 'LOCAL'}";
}
boolean isExperimental = cli['bintrayExperimental'].or(false).boolean
if(isExperimental) {
  return "EXPERIMENTAL-${System.getenv('BUILD_NUMBER') ?: 'LOCAL'}";
}
return version;
```

I was just passing through, thought I'd add what I think reads easier